lib/socketpair.c

```diff
@@ -37,6 +37,12 @@
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
 #endif
+#ifdef __AMIGA__
+#include <netinet/in.h> /* IPPROTO_TCP */
+#ifndef INADDR_LOOPBACK
+#define INADDR_LOOPBACK (u_long)0x7f000001
+#endif /* !INADDR_LOOPBACK */
+#endif /* __AMIGA__ */
```

Comment:
I realize you did this change with "minimal impact" considerations for other systems, but I would still suggest that we expand it somewhat to make it a fix that potentially also fix the build for other older-style includes and defines. What would you say about this?
~~~diff
--- a/lib/socketpair.c
+++ b/lib/socketpair.c
@@ -35,10 +35,16 @@
 #include <io.h>
 #else
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
 #endif
+#ifdef HAVE_NETINET_IN_H
+#include <netinet/in.h> /* IPPROTO_TCP */
+#endif
+#ifndef INADDR_LOOPBACK
+#define INADDR_LOOPBACK 0x7f000001
+#endif /* !INADDR_LOOPBACK */
 #endif /* !WIN32 */
 
 /* The last 3 #include files should be in this order */
 #include "curl_printf.h"
 #include "curl_memory.h"

Reply:
Looks good to me.

Reply:
Can you force-push that version so that we get that through the CI just to make sure?